src/capi/azext_capi/custom.py

```diff
@@ -706,17 +707,23 @@ def get_kubeconfig(capi_name):
     return f"Wrote kubeconfig file to {filename} "
 
 
-def delete_workload_cluster(cmd, capi_name, yes=False):
+def delete_workload_cluster(cmd, capi_name=None, resource_group_name=None, yes=False):
+    if not capi_name and not resource_group_name:
+        raise RequiredArgumentMissingError("Missing --name/-n or--resource-group/-g for self-managed clusters")
+    if capi_name and resource_group_name:
+        raise MutuallyExclusiveArgumentError("--name/-n and --resource-group/-g are mutally exclusive")
     exit_if_no_management_cluster()
     msg = f'Do you want to delete this Kubernetes cluster "{capi_name}"?'
+    command = ["kubectl", "delete", "cluster", capi_name]
+    if resource_group_name:
```

Comment:
I'm not sure I fully understand how this works, how do we tell if the management cluster is the same cluster as the workload cluster?

Reply:
That seems a bit too subtle to me as a UX, what if the user sees the -g option and passes it in for a workload cluster that isn't self-managed?

Is there any other way we can detect if the cluster is managing itself?